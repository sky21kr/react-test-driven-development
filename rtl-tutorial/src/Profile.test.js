import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const utils = render(<Profile username="sky21kr" name="신승민"/> )
        expect(utils.container).toMatchSnapshot();
    });
    it('shows the props correctly', () => {
        const utils = render(<Profile username="sky21kr" name="신승민"/>)
        utils.getByText('sky21kr');
        utils.getByText('(신승민)');
        utils.getByText(/신/);
    })
})