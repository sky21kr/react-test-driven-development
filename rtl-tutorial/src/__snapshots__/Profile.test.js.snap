// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      sky21kr
    </b>
     
    <span>
      (
      신승민
      )
    </span>
  </div>
</div>
`;
